cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -129,6 +130,14 @@ def remove_implied_values(self, data, **kwargs):
                 value[:] = [v for v in value if not _is_implied(v)]
         return data
 
+    @pre_dump
+    def unwrap_marked_class(self, data, **kwargs):
+        """Remove value implied by the code. i.e., only keep parameters that were specified in the yaml file."""
+        for key, value in vars(data).items():
+            if isinstance(value, Param):
+                setattr(data, key, value.value)
```

Comment:
value.value is not so nice.

let's use param_name, param_value vs key, value

Reply:
Yea, so weird. But before `isinstance(value, Param)`, we can't be sure if we can call it `param`.
In other words, I would prefer wording weirdness than logic weirdness